internal/backend/remote-state/http/backend.go

```diff
@@ -132,46 +132,42 @@ type Backend struct {
 
 // configureTLS configures TLS when needed; if there are no conditions requiring TLS, no change is made.
 func (b *Backend) configureTLS(client *retryablehttp.Client, data *schema.ResourceData) error {
-	// If there are no conditions needing to configure TLS, leave the client untouched
-	skipCertVerification := data.Get("skip_cert_verification").(bool)
-	clientCACertificatePem := data.Get("client_ca_certificate_pem").(string)
-	clientCertificatePem := data.Get("client_certificate_pem").(string)
-	clientPrivateKeyPem := data.Get("client_private_key_pem").(string)
-	if !skipCertVerification && clientCACertificatePem == "" && clientCertificatePem == "" && clientPrivateKeyPem == "" {
-		return nil
-	}
-	if clientCertificatePem != "" && clientPrivateKeyPem == "" {
-		return fmt.Errorf("client_certificate_pem is set but client_private_key_pem is not")
-	}
-	if clientPrivateKeyPem != "" && clientCertificatePem == "" {
-		return fmt.Errorf("client_private_key_pem is set but client_certificate_pem is not")
-	}
+    // If there are no conditions needing to configure TLS, leave the client untouched
+    skipCertVerification := data.Get("skip_cert_verification").(bool)
+    clientCACertificatePem := data.Get("client_ca_certificate_pem").(string)
+    clientCertificatePem := data.Get("client_certificate_pem").(string)
+    clientPrivateKeyPem := data.Get("client_private_key_pem").(string)
+    if !skipCertVerification && clientCACertificatePem == "" && clientCertificatePem == "" && clientPrivateKeyPem == "" {
+        return nil
+    }
+    if clientCertificatePem != "" && clientPrivateKeyPem == "" {
+        return fmt.Errorf("client_certificate_pem is set but client_private_key_pem is not")
+    }
+    if clientPrivateKeyPem != "" && clientCertificatePem == "" {
+        return fmt.Errorf("client_private_key_pem is set but client_certificate_pem is not")
+    }
 
-	// TLS configuration is needed; create an object and configure it
-	var tlsConfig tls.Config
-	client.HTTPClient.Transport.(*http.Transport).TLSClientConfig = &tlsConfig
+    // TLS configuration is needed; create an object and configure it
+    var tlsConfig tls.Config
+    client.HTTPClient.Transport.(*http.Transport).TLSClientConfig = &tlsConfig
 
-	if skipCertVerification {
-		// ignores TLS verification
-		tlsConfig.InsecureSkipVerify = true
```

Comment:
If I understand correct, this `tlsConfig.InsecureSkipVerify = true` assignment is removed in this PR. Do we want to completely remove this functionality where the practitioner might prefer to skip the TLS verification for various purposes (testing, development etc.)? 

Also, if we decide to remove this functionality to be able to skip the verification, it needs to be added to the Changelog, If I'm not mistaken.